vllm/worker/model_runner.py

```diff
@@ -869,11 +905,31 @@ def _maybe_cupy_nccl():
         yield
 
 
+def _pad_to_alignment(x: List[int], multiple_of: int, pad: int) -> List[int]:
+    return x + [pad] * ((-len(x)) % multiple_of)
+
+
 def _pad_to_max(x: List[int], max_len: int, pad: int) -> List[int]:
     assert len(x) <= max_len
     return x + [pad] * (max_len - len(x))
 
 
+def _make_tensor_with_pad_for_alignment(
+    x: List[int],
+    pad: int,
+    dtype: torch.dtype,
+    device: Optional[Union[str, torch.device]],
+) -> torch.Tensor:
+    """Create a tensor of a given list x with padding.
+    It adds paddings to align with graph batch size. See
+    _get_graph_batch_size for more details.
+    """
+    batch_size = len(x)
+    batch_size = _get_graph_batch_size(batch_size)
+    padded_x = _pad_to_alignment(x, batch_size, pad)
+    return torch.tensor(padded_x, dtype=dtype, device=device)
```

Comment:
I believe we should decouple this from `graph_batch_size` even if we want to add paddings in eager mode. Can we have `target_batch_size` as an input parameter?

Reply:
to make sure I understand, we want to decouple graph batch size and just batch size for padding right?

Reply:
on it...